internal/controller/accessrequest_controller.go

```diff
@@ -316,11 +359,32 @@ func (r *AccessRequestReconciler) findObjectsForRoleTemplate(ctx context.Context
 	return requests
 }
 
-// findObjectsForProject will retrieve all AccessRequest resources referencing
+// findAccessRequestsByUserApp will list all AccessRequests in the given namespace
+// filtering by the given username, appName and appNamespace.
+func (r *AccessRequestReconciler) findAccessRequestsByUserApp(ctx context.Context, namespace, username, appName, appNamespace string) (*api.AccessRequestList, error) {
```

Comment:
```suggestion
func (r *AccessRequestReconciler) findAccessRequestsByUserAndApp(ctx context.Context, namespace, username, appName, appNamespace string) (*api.AccessRequestList, error) {
```

I dislike long function names, but I think this would be significantly clearer.

Reply:
done